.github/workflows/build-and-test.yml

```diff
@@ -19,3 +19,6 @@ jobs:
       - run: npm ci
       - run: npm run build
       - run: npm test
+
+      - name: Jest Coverage Report
```

Comment:
I would have created another config file, something like
```
name: Jest Coverage Report

on: [pull_request]

jobs:
  jest-coverage-report:

    runs-on: ubuntu-latest

    steps:
      - name: Checkout Repository
        uses: actions/checkout@v4

      - name: Coverage Report
        uses: ArtiomTr/jest-coverage-report-action@v2  
```

Reply:
I had to add this to the step:

```      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version: '20.x'
```

as the coverage report was failing to build the report without it

Reply:
It looks good to me 👍, change the node action to latest version `actions/setup-node@v4`